go/tasks/plugins/k8s/sagemaker/hpo_job.go

```diff
@@ -0,0 +1,319 @@
+
+	hpoJob := &hpojobv1.HyperparameterTuningJob{
+		Spec: hpojobv1.HyperparameterTuningJobSpec{
+			HyperParameterTuningJobName: &taskName,
+			HyperParameterTuningJobConfig: &commonv1.HyperParameterTuningJobConfig{
+				ResourceLimits: &commonv1.ResourceLimits{
+					MaxNumberOfTrainingJobs: awssagemaker.ToInt64Ptr(sagemakerHPOJob.GetMaxNumberOfTrainingJobs()),
+					MaxParallelTrainingJobs: awssagemaker.ToInt64Ptr(sagemakerHPOJob.GetMaxParallelTrainingJobs()),
+				},
+				Strategy: getAPIHyperParameterTuningJobStrategyType(hpoJobConfig.GetTuningStrategy()),
+				HyperParameterTuningJobObjective: &commonv1.HyperParameterTuningJobObjective{
+					Type:       getAPIHyperparameterTuningObjectiveType(hpoJobConfig.GetTuningObjective().GetObjectiveType()),
+					MetricName: awssagemaker.ToStringPtr(hpoJobConfig.GetTuningObjective().GetMetricName()),
+				},
+				ParameterRanges:              hpoJobParameterRanges,
+				TrainingJobEarlyStoppingType: getAPITrainingJobEarlyStoppingType(hpoJobConfig.TrainingJobEarlyStoppingType),
+			},
+			TrainingJobDefinition: &commonv1.HyperParameterTrainingJobDefinition{
+				StaticHyperParameters: staticHyperparams,
+				AlgorithmSpecification: &commonv1.HyperParameterAlgorithmSpecification{
+					TrainingImage:     awssagemaker.ToStringPtr(trainingImageStr),
+					TrainingInputMode: getAPITrainingInputMode(sagemakerHPOJob.GetTrainingJob().GetAlgorithmSpecification().GetInputMode()),
+				},
+				InputDataConfig: []commonv1.Channel{
+					{
+						ChannelName: awssagemaker.ToStringPtr("train"),
+						DataSource: &commonv1.DataSource{
+							S3DataSource: &commonv1.S3DataSource{
+								S3DataType: "S3Prefix",
+								S3Uri:      awssagemaker.ToStringPtr(trainPathLiteral.GetScalar().GetBlob().GetUri()),
+							},
+						},
+						ContentType: awssagemaker.ToStringPtr("text/csv"), // TODO: can this be derived from the task spec?
+						InputMode:   "File",
+					},
+					{
+						ChannelName: awssagemaker.ToStringPtr("validation"),
+						DataSource: &commonv1.DataSource{
+							S3DataSource: &commonv1.S3DataSource{
+								S3DataType: "S3Prefix",
+								S3Uri:      awssagemaker.ToStringPtr(validatePathLiteral.GetScalar().GetBlob().GetUri()),
+							},
+						},
+						ContentType: awssagemaker.ToStringPtr("text/csv"), // TODO: can this be derived from the task spec?
+						InputMode:   "File",
+					},
+				},
+				OutputDataConfig: &commonv1.OutputDataConfig{
+					S3OutputPath: awssagemaker.ToStringPtr(outputPath),
+				},
+				ResourceConfig: &commonv1.ResourceConfig{
+					InstanceType:   sagemakerHPOJob.GetTrainingJob().GetTrainingJobConfig().GetInstanceType(),
+					InstanceCount:  awssagemaker.ToInt64Ptr(sagemakerHPOJob.GetTrainingJob().GetTrainingJobConfig().GetInstanceCount()),
+					VolumeSizeInGB: awssagemaker.ToInt64Ptr(sagemakerHPOJob.GetTrainingJob().GetTrainingJobConfig().GetVolumeSizeInGb()),
+					VolumeKmsKeyId: awssagemaker.ToStringPtr(""), // TODO: add to proto and flytekit
```

Comment:
interesting, this is to support encrypted data i guess

Reply:
Yes it is. I haven't added that to flytekit yet because I am not sure whether we should allow users to encrypt